w1r3/go/main.go

```diff
@@ -309,19 +316,25 @@ func cpuElapsed(start time.Duration) (time.Duration, error) {
 	return totalCpu(end) - start, nil
 }
 
-func uploadStep(ctx context.Context, tracer trace.Tracer,
-	latency metric.Float64Histogram,
-	cpu metric.Float64Histogram,
-	commonAttributes []attribute.KeyValue,
-	config Config,
+type StepConfig struct {
+	tracer           trace.Tracer
+	latency          metric.Float64Histogram
+	cpu              metric.Float64Histogram
+	commonAttributes []attribute.KeyValue
+}
+
+func uploadStep(ctx context.Context,
+	config BenchmarkConfig,
+	stepConfig StepConfig,
```

Comment:
I think it's more typical to pass config structs around by pointer rather than value. It is fine for now (since the fields are interfaces and slices) but gets messy if e.g. there are nested structs that then get copied.

Reply:
Ack